tests/unit/test_functional_post_install.py

```diff
@@ -355,3 +355,20 @@ def test_cylc_no_rose(tmp_path):
 def test_copy_config_file_fails(tmp_path):
     """It fails if source not a rose suite."""
     copy_config_file(tmp_path, tmp_path)
+
+
+def test_CLI_opts_cleared(monkeypatch):
+    """The post install entry point should clear the three rose opts"""
+    path = 'cylc.rose.entry_points.'
+    monkeypatch.setattr(
+        f'{path}record_cylc_install_options', lambda *_, **__: False)
+    monkeypatch.setattr(f'{path}copy_config_file', lambda *_, **__: False)
+    monkeypatch.setattr(f'{path}rose_config_exists', lambda *_, **__: True)
+
+    opts = SimpleNamespace(
+        rose_template_vars=['gee', 'whiz'],
+        defines=['foo', 'bar'],
+        opt_conf_keys=['thomas', 'percy'])
+
+    post_install('', '', opts)
+    assert any(opts.__dict__.values()) is False
```

Comment:
For a functional test, how about something installs and then reinstalls a workflow using the same `opts` object e.g:

```python
def test_something(cylc_install_cl):
    opts = SimpleNamespace(
        rose_template_vars=['gee', 'whiz'],
        defines=['foo', 'bar'],
        opt_conf_keys=['thomas', 'percy']
    )

    # install
    run_dir = cylc_install_cli(..., opts)

    # check installed rose config
    RoseConfigTree(run_dir).get(('template variables', 'gee')) == 'whatever'

    # reinstall using same opts object (just like cylc-flow does)
    run_dir = cylc_install_cli(..., opts)

     # check the installed rose config again
    RoseConfigTree(run_dir).get(('template variables', 'gee')) == 'whatever'
```

Reply:
>     # reinstall using same opts object (just like cylc-flow does)

Sorry, but Cylc Flow doesn't reinstall with the same opts object as Cylc install - do you mean play?

Reply:
I don't understand how this is supposed to demonstate the broken state before my changes. `cylc reinstall` interacts with the Opts and the source directory (the validate step in VR does the interaction with the run dir). In this bug the problem is the interaction between the opts and the run directory when `cylc play` is run. I'm happy to do _that_ as a functional test.

Reply:
I don't think that this makes sense: the problem was with the content of the run directory interacting with `cylc play`, not the source directory. I'll translate my manual test into one of these.